applications/client/modules/procedureapi.py

```diff
@@ -6,18 +6,44 @@ class ProcedureApi():
     def __init__(self, procedure_name):
         self.procedure_name = procedure_name
 
-    def write_value(self, name, data):
-        pass
+    def write_value(self, dictionary):
+        """ Writes key value pairs into the values table
+        param dictionary: Takes dictionary of key/value pairs as input
+        """
+        db = current.db
+        for key in dictionary.keys():
+            val = dictionary[key]
```

Comment:
You can iterate through dict directly.

Basically the two lines can be replaced by:

for key, val in dictionary.iteritems():